contracts/SSVNetwork.sol

```diff
@@ -239,413 +239,271 @@ contract SSVNetwork is OwnableUpgradeable, ISSVNetwork {
-
-            if (validator.active) {
-                ++activeValidatorCount;
+        {
+            if (!pod.disabled) {
+                DAO memory dao = _dao;
+                dao = _updateDAOEarnings(dao);
+                --dao.validatorCount;
+                _dao = dao;
             }
-            // Changing to a single event reducing by 15K gas
         }
-        emit BulkValidatorTransferred(publicKeys, toClusterId, shares);
-
-        uint64[] memory oldOperatorIds = _clusters[fromClusterId].operatorIds;
-        uint64[] memory newOperatorIds = _clusters[toClusterId].operatorIds;
-
-        _updateOperatorsOnTransfer(oldOperatorIds, newOperatorIds, activeValidatorCount);
-
-        Pod memory pod = _pods[keccak256(abi.encodePacked(msg.sender, fromClusterId))];
-        uint64 podIndex = _clusterCurrentIndex(fromClusterId);
-        pod.usage.balance = _podBalance(pod, podIndex);
-        pod.usage.index = podIndex;
-        pod.usage.block = uint64(block.number);
-        pod.validatorCount -= activeValidatorCount;
-
-        _pods[keccak256(abi.encodePacked(msg.sender, fromClusterId))] = pod;
-
-        pod = _pods[keccak256(abi.encodePacked(msg.sender, toClusterId))];
-        podIndex = _clusterCurrentIndex(toClusterId);
-        pod.usage.balance = _podBalance(pod, podIndex);
-        pod.usage.index = podIndex;
-        pod.usage.block = uint64(block.number);
-        pod.validatorCount += activeValidatorCount;
-
-        _pods[keccak256(abi.encodePacked(msg.sender, toClusterId))] = pod;
-
-        if (_liquidatable(pod.disabled, _podBalance(pod, podIndex), pod.validatorCount, newOperatorIds)) {
-            revert PodLiquidatable();
-        }
-    }
-    */
-
-    /**************************/
-    /* Pod External Functions */
-    /**************************/
-
-    /*
-    function registerPod(uint64[] memory operatorIds, uint256 amount) external override {
-        _validateOperatorIds(operatorIds);
-
-        bytes32 clusterId = keccak256(abi.encodePacked(operatorIds));
-
-        if (_clusters[clusterId].operatorIds.length == 0) {
-            _createClusterUnsafe(clusterId, operatorIds);
-        }
-
-        bytes32 hashedPod = keccak256(abi.encodePacked(msg.sender, clusterId));
-
-        if (_pods[hashedPod].usage.block != 0) {
-            revert PodAlreadyExists();
-        }
-
-        _pods[hashedPod].usage.block = uint64(block.number);
+        delete _validatorPKs[hashedValidator];
 
-        emit PodCreated(msg.sender, clusterId);
+        _pods[hashedPod] = keccak256(abi.encodePacked(pod.validatorCount, pod.networkFee, pod.networkFeeIndex, pod.index, pod.balance, pod.disabled ));
 
-        _deposit(msg.sender, clusterId, amount.shrink());
+        emit ValidatorRemoved(msg.sender, operatorIds, publicKey);
+        emit PodMetadataUpdated(msg.sender, operatorIds, pod);
     }
-    */
 
-    function liquidate(
-        address ownerAddress,
+    function liquidatePod(
```

Comment:
Maybe there should be a validation of pod.disabled with revert of podAlreadyLiquidated?

I get that it will revert with _liquidatable check, just a suggestion

Reply:
@andrew-blox no that's not. we have validation in function _validatePodIsNotLiquidated(pod);